components/mpas-albany-landice/src/shared/mpas_li_time_average_coupled.F

```diff
@@ -0,0 +1,135 @@
+        call mpas_pool_get_array(timeAveragingPool, 'avgCalvingFlux', avgCalvingFlux)
+        call mpas_pool_get_array(timeAveragingPool, 'avgFaceMeltFlux', avgFaceMeltFlux)
+
+        do iCell = 1, nCells
+           avgBareIceAblationApplied(iCell) = 0.0_RKIND
+           avgCalvingFlux(iCell) = 0.0_RKIND
+           avgFaceMeltFlux(iCell) = 0.0_RKIND
+        end do
+
+        timeAccumulatedCoupled = 0.0_RKIND
+
+    end subroutine li_time_average_coupled_init
+
+!***********************************************************************
+!
+!  routine li_time_average_coupled_accumulate
+!
+!> \brief   Coupled time averager accumulation
+!> \author  Stephen Price
+!> \date    04 March 2025
+!> \details
+!>  This routine accumulates the coupled time averaging fields
+!
+!-----------------------------------------------------------------------
+    subroutine li_time_average_coupled_accumulate(timeAveragingPool, geometryPool, meshPool)
+            
+        use li_setup 
+
+        type (mpas_pool_type), intent(inout) :: timeAveragingPool
+        type (mpas_pool_type), intent(in) :: geometryPool
+        type (mpas_pool_type), intent(in) :: meshPool
+
+        real (kind=RKIND), dimension(:), pointer :: bareIceAblationApplied, avgBareIceAblationApplied, &
+                                                    calvingThickness, avgCalvingFlux, &
+                                                    faceMeltingThickness, avgFaceMeltFlux
+        integer, pointer :: nCells
+
+        real (kind=RKIND), pointer :: timeAccumulatedCoupled, rhoi, deltat
+
+        integer :: iCell
+
+        call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+        call mpas_pool_get_config(liConfigs, 'config_ice_density', rhoi)
+        call mpas_pool_get_array(meshPool, 'deltat', deltat)
+        call mpas_pool_get_array(timeAveragingPool, 'timeAccumulatedCoupled', timeAccumulatedCoupled)
+        call mpas_pool_get_array(geometryPool, 'bareIceAblationApplied', bareIceAblationApplied)
+        call mpas_pool_get_array(geometryPool, 'calvingThickness', calvingThickness)
+        call mpas_pool_get_array(geometryPool, 'faceMeltingThickness', faceMeltingThickness)
+        call mpas_pool_get_array(timeAveragingPool, 'avgBareIceAblationApplied', avgBareIceAblationApplied)
+        call mpas_pool_get_array(timeAveragingPool, 'avgCalvingFlux', avgCalvingFlux)
+        call mpas_pool_get_array(timeAveragingPool, 'avgFaceMeltFlux', avgFaceMeltFlux)
+
+        do iCell = 1, nCells
+
+           avgBareIceAblationApplied(iCell) = ( avgBareIceAblationApplied(iCell) * timeAccumulatedCoupled &
+                                               + bareIceAblationApplied(iCell) * deltat ) / ( timeAccumulatedCoupled + deltat )
+
+           avgCalvingFlux(iCell) = ( avgCalvingFlux(iCell) * timeAccumulatedCoupled &
+                                                + calvingThickness(iCell) * deltat * (rhoi / deltat) ) / ( timeAccumulatedCoupled + deltat )
```

Comment:
Why is calvingThickness multiplied by deltat and then immediately divided by deltat here?

Reply:
We could just leave out the delta t in the num and denom ... but Matt and I thought that leaving it in might be more explicit, e.g. to clarify that "rhoi / delta" is the conversion factor that you need to go from a calving thickness (in m) to a calving flux (in kg * m^2 / s). We could remove if this is too confusing.

Reply:
Okay, as long as it was intentional, that's fine. I was just worried that the parentheses had gotten misaligned or something.